### PR TITLE
[Edit profile, My habits] "User credo" text is displayed if user doesn't fill 'Credo' field #2639

### PR DESCRIPTION
**Pull request by Edw1x**
# Problem:
## User credo" text is displayed if the user doesn't fill the 'Credo' field
image

# Actual result:
image
image

# Expected result: 
## 'Credo' should not be exist

# Result: 
### No 'Credo' field
image
### 'Credo' field exists
image

### CODE REVIEW

File: src/app/component/user/components/profile/profile-widget/profile-header/profile-header.component.html [hunk 1]
```diff
@@ -23,7 +23,9 @@
     <div class="rate">
       <p>{{ 'profile.rate' | translate }}: {{ userInfo.rating || mockedUserInfo.rating }}</p>
     </div>
-    <div class="credo">{{ userInfo.userCredo || mockedUserInfo.userCredo }}</div>
+    <div *ngIf="userInfo.userCredo.length > 0"
+          class="credo">{{ userInfo.userCredo || mockedUserInfo.userCredo }}
+    </div>
```
Comment (yuriiLisunov) [inline src/app/component/user/components/profile/profile-widget/profile-header/profile-header.component.html:28]: you can check "userInfo.userCredo?.length"

Comment (Edw1x) [reply src/app/component/user/components/profile/profile-widget/profile-header/profile-header.component.html]: changed

**Review by idzhersh — APPROVED** [review]

**Review by yuriiLisunov — APPROVED** [review]

**Review by VitalyBorysenko — APPROVED** [review]

File: src/app/component/user/components/profile/profile-widget/profile-header/profile-header.component.html [hunk 2]
```diff
@@ -23,7 +23,9 @@
     <div class="rate">
       <p>{{ 'profile.rate' | translate }}: {{ userInfo.rating || mockedUserInfo.rating }}</p>
     </div>
-    <div class="credo">{{ userInfo.userCredo || mockedUserInfo.userCredo }}</div>
+    <div *ngIf="userInfo.userCredo?.length"
```
Comment (JimHolman1199) [inline src/app/component/user/components/profile/profile-widget/profile-header/profile-header.component.html:26]: what if userInfo will be undefined ? what than "*ngIf="undefined"" will produce ? error ?

Comment (Edw1x) [reply src/app/component/user/components/profile/profile-widget/profile-header/profile-header.component.html:26]: So, you mean to change this line of code to 'userInfo.userCredo?.length && userInfo != undefined' ?

Comment (JimHolman1199) [reply src/app/component/user/components/profile/profile-widget/profile-header/profile-header.component.html:26]: i think we need to move it getter and do a check there... 
get something () { 
if (userInfo && userInfo.userCredo) {return userInfo.userCredo.length}  
} return 0

Comment (Edw1x) [reply src/app/component/user/components/profile/profile-widget/profile-header/profile-header.component.html:26]: Roger that

**Review by KirillAbramo — APPROVED** [review]

File: src/app/component/user/components/profile/profile-widget/profile-header/profile-header.component.ts [hunk 3]
```diff
@@ -30,6 +30,13 @@ export class ProfileHeaderComponent implements OnInit, OnDestroy {
     this.buildSocialNetworksChart();
   }
 
+  get checkUserCredo() {
```
Comment (JimHolman1199) [inline src/app/component/user/components/profile/profile-widget/profile-header/profile-header.component.ts:33]: please add type of returned value 
get checkUserCredo(): number {....

Comment (Edw1x) [reply src/app/component/user/components/profile/profile-widget/profile-header/profile-header.component.ts]: Done

**Review by JimHolman1199 — APPROVED** [review]

**Pull request merged**